boot/bootutil/src/encrypted.c

```diff
@@ -305,7 +309,14 @@ boot_enc_load(const struct image_header *hdr, const struct flash_area *fap,
 int
 boot_enc_valid(const struct flash_area *fap)
 {
-    return enc_state[fap->fa_id - FLASH_AREA_IMAGE_PRIMARY].valid;
+    int rc;
+
+    rc = flash_area_id_to_image_slot(fap->fa_id);
+    if (rc < 0) {
+        return rc;
```

Comment:
`boot_enc_valid` is used as bool (maybe it should return bool!), so this would be true?

Reply:
I think it would be safer to return `0` here, that would just skip trying encrypted stuff and postpone the error for a later function...

Reply:
You are right -  `boot_enc_valid` is used as it returns boolean. I Will change its type.